rep-2004.rst

```diff
@@ -0,0 +1,527 @@
+REP: 2004
+Title: Package Quality Categories
+Author: William Woodall <william@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 17-Dec-2019
+Post-History: 17-Dec-2019
+
+
+Abstract
+========
+
+This REP describes a set of categories which are meant to convey the quality, or at least the maturity, of packages in the ROS ecosystem.
+Inclusion in one category or another is based on the policies to which the packages adhere.
+The categories are meant to give some expectation as to the quality of a package and allows the maintainers to be more strict with some packages and less so with others.
+
+The purpose of these categories is not to enforce quality, but to set expectations for consumers of the packages and to encourage maintainers of the packages to document how their package's policies achieve that quality level.
```

Comment:
The package maintainers get to claim and justify their quality level, this document does not set in place a certification process, certification regime, or proposed automated system(s) that _enforces_ that packages only claim levels that they meet.

The best this can offer is "this package claims level X with this justification" and the user can evaluate the justification themselves or look for a peer-reviewed list of level X packages to see if this package is on that list. At which point they could trust the peer-review process, but there's nothing in this document that ensures packages are following the rules.

Reply:
Maybe it could be reworded slightly?

> The purpose of these categories is to establish guidelines for compliance with various quality levels.  It is up to the package maintainers to establish processes and document their compliance with a given category.